keps/sig-cluster-lifecycle/kubeadm/3929-no-cri-socket-annotation/README.md

```diff
@@ -0,0 +1,772 @@
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md)
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+The proposal is to remove the Container Runtime Interface (CRI) socket annotation
+from the Node object in Kubernetes, which is currently added during the
+"init upload-config" phase in Kubeadm. This annotation is used to specify the CRI
+socket endpoint used by the kubelet on each node for communication with the container
+runtime. Instead of relying on this annotation, the proposal suggests using a global
+kubelet configuration with a CRI socket specified, as well as providing the ability to
+override this configuration during kubeadm join using the --config flag. This would
+eliminate the need for kubeadm to store CRI socket configuration on each node, and
+instead rely on the Kubernetes configuration files for specifying this information.
+
+## Motivation
+
+Currently, kubeadm adds a CRI socket annotation to the Node object during the
+"init upload-config" phase, which specifies the endpoint for the CRI that is being
+used by the kubelet on each node. This annotation is persistent on the Node object,
+even if the kubelet is updated or the CRI is changed.
+
+After migration of container runtime endpoint flag to kubelet config, we can set
+cri socket in kubelet configuration.
+
+### Goals
+
+- Remove the use of CRI socket annotation on Node object
+- For node customized kubelet configuration, it can be saved locally on disk with file path
+  `/var/lib/kubelet/kubeadm-config.yaml` and we will prioritize respecting the local
```

Comment:
it's one of the ways of doing it. we need to be careful and make sure this host / instance config does not cause issues for us in the future.

perhaps we can avoid it, and just parse the cri value from the existing config.yaml on upgrade. like i proposed on the linked issue.